doc/source/fq_nmod_sparse_mat.rst

```diff
@@ -0,0 +1,295 @@
+    entries per row, with individual entries generated by fq_nmod_randtest
+
+
+Output
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_mat_print_pretty(const fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Prints the matrix ``M`` to ``stdout`` in a human-readable format
+
+
+Comparison
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_is_zero(fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Checks if the given matrix ``M`` is trivial (empty), returning `1` if so and `0` 
+    otherwise
+
+.. function:: void fq_nmod_sparse_mat_equal(const fq_nmod_sparse_mat_t M1, const fq_nmod_sparse_mat_t M2, slong ioff, const fq_nmod_ctx_t ctx)
+
+    Checks if ``M1`` equals ``M2``, returning `1` if so and `0` otherwise
+
+
+Transpose
+--------------------------------------------------------------------------------
+
+.. function:: void fq_sparse_mat_transpose(fq_nmod_sparse_mat_t N, const fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Transposes ``M`` into the matrix ``N`` (must have swapped rows and columns)
+
+
+Arithmetic
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_mat_neg(fq_nmod_sparse_mat_t N, const fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Sets ``N`` to the negation of ``M``
+
+.. function:: void fq_nmod_sparse_mat_scalar_mul_fq_nmod(fq_nmod_sparse_mat_t N, const fq_nmod_sparse_mat_t M, const fq_nmod_t c, const fq_nmod_ctx_t ctx)
+
+    Sets ``N`` to the scalar multiple of ``M`` by ``c``
+
+.. function:: void fq_nmod_sparse_mat_add(fq_nmod_sparse_mat_t O, const fq_nmod_sparse_mat_t M, const fq_nmod_sparse_mat_t N, const fq_nmod_ctx_t ctx)
+
+    Sets ``O`` to the sum of ``M`` and ``N``
+
+.. function:: void fq_nmod_sparse_mat_sub(fq_nmod_sparse_mat_t O, const fq_nmod_sparse_mat_t M, const fq_nmod_sparse_mat_t N, const fq_nmod_ctx_t ctx)
+
+    Sets ``O`` to the difference of ``M`` and ``N``
+
+.. function:: void fq_nmod_sparse_mat_scalar_addmul_fq_nmod(fq_nmod_sparse_mat_t O, const fq_nmod_sparse_mat_t M, const fq_nmod_sparse_mat_t N, const fq_nmod_t c, const fq_nmod_ctx_t ctx)
+
+    Sets ``O`` to the sum of ``M`` and ``c` times ``N``
+
+.. function:: void fq_nmod_sparse_mat_scalar_submul_fq_nmod(fq_nmod_sparse_mat_t O, const fq_nmod_sparse_mat_t M, const fq_nmod_sparse_mat_t N, const fq_nmod_t c, const fq_nmod_ctx_t ctx)
+
+    Sets ``O`` to the difference of ``M`` and ``N` times ``v``
+
+.. function:: void fq_nmod_sparse_mat_mul_vec(fq_nmod_struct *y, const fq_nmod_sparse_mat_t M, const fq_nmod_struct *x, const fq_nmod_ctx_t ctx)
+
+    Sets ``y`` to the product of ``M`` and ``x``
+
+.. function:: void fq_nmod_sparse_mat_mul_mat(fq_nmod_mat_t Y, const fq_nmod_sparse_mat_t M, const fq_nmod_mat_t X, const fq_nmod_ctx_t ctx)
```

Comment:
The naming convention of this function does not match that of the fq_nmod_mat module. You should omit the _mat at the end.